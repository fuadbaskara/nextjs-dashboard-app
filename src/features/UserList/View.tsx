/* eslint-disable operator-linebreak */
import React from 'react';
import Link from 'next/link';
import Table from '@/components/common/Table';
import { columns } from './components/columns';
import useHomeState from './ViewModel';
import Filter from './components/filter';

function UserList() {
  const {
    data,
    page,
    filter,
    loading,
    searchValue,
    onSearch,
    onChangeGender,
    onChangePagination,
    onResetFilter,
    onSort,
  } = useHomeState();

  return (
    <>
      <Link href="/">
        <span className="hover:text-blue-400">Home</span>
      </Link>
      <span className="px-2">/</span>
      <h2 className="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl">
        User List
      </h2>
      <Filter
        gender={filter.gender}
        searchValue={searchValue}
        onSearch={onSearch}
        onSelect={onChangeGender}
        onResetFilter={onResetFilter}
      />
      <Table
        dataSource={data}
        columns={columns({ onSort })}
        loading={loading}
        rowKey="cell"
        pagination={{
          current: page,
          onChange: (currPage) => onChangePagination(currPage),
        }}
      />
    </>
  );
}

export default UserList;
