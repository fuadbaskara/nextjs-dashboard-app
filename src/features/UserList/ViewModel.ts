import { useCallback, useEffect, useState } from 'react';
import { debounce } from 'lodash';
import { useAppDispatch, useAppSelector } from '@/hooks';
import { getUsersData, userListSelector } from '@/store/slices/users';
import {
  filterSelector,
  onChangeSearch,
  onChangeGender,
  onChangePagination,
  onResetFilter,
  onChangeSort,
} from '@/store/slices/filter';
import { SortOrder } from '@/api/models';

const useHomeState = () => {
  const dispatch = useAppDispatch();
  const { data, info, loading } = useAppSelector(userListSelector);
  const { filter } = useAppSelector(filterSelector);
  const [searchValue, setSearchValue] = useState('');

  const fetchUserList = (params?: any) => {
    dispatch(getUsersData({ params }));
  };

  const onSearch = (keyword: string) => {
    dispatch(onChangeSearch({ keyword }));
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const debounceSearch = useCallback(debounce(onSearch, 500), []);

  const onDebounceSearch = (keyword: string) => {
    setSearchValue(keyword);
    debounceSearch(keyword.toLowerCase());
  };

  const onSort = (sortBy: string, sortOrder: SortOrder) => {
    dispatch(onChangeSort({ sortBy, sortOrder }));
  };

  const onPaginationChange = (page: number) => {
    dispatch(onChangePagination({ page }));
  };

  const onGenderChange = (value: string) => {
    const gender = value as '' | 'male' | 'female';
    dispatch(onChangeGender({ gender }));
  };

  const onFilterReset = () => {
    setSearchValue('');
    dispatch(onResetFilter());
  };

  useEffect(() => {
    fetchUserList(filter);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [filter]);

  return {
    data,
    page: filter.page,
    filter,
    info,
    loading,
    searchValue,
    onSearch: onDebounceSearch,
    onChangeGender: onGenderChange,
    onChangePagination: onPaginationChange,
    onResetFilter: onFilterReset,
    onSort,
  };
};

export default useHomeState;
