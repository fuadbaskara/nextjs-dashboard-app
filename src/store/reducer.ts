import { combineReducers } from '@reduxjs/toolkit';

// all reducers
import usersReducer from './slices/users';
import filterReducer from './slices/filter';

const rootReducer = combineReducers({
  users: usersReducer,
  filter: filterReducer,
});

export default rootReducer;
