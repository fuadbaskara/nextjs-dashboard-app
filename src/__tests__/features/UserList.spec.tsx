import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../store';
import UserList from '../../features/UserList/View';

const renderUserList = () =>
  render(
    <Provider store={store}>
      <UserList />
    </Provider>,
  );

describe('Render UserList', () => {
  test('render UserList to match snapshot:', () => {
    const { asFragment } = renderUserList();
    expect(asFragment()).toMatchSnapshot();
  });
});
