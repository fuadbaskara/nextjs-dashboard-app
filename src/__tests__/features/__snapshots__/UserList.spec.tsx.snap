// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserList render UserList to match snapshot: 1`] = `
<DocumentFragment>
  <span
    class="hover:text-blue-400"
  >
    Home
  </span>
  <span
    class="px-2"
  >
    /
  </span>
  <h2
    class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl"
  >
    User List
  </h2>
  <div
    class="flex flex-col space-y-4 md:flex-row md:items-end md:space-y-0 md:space-x-4"
  >
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="search"
      >
        Search
      </label>
      <div
        class="mt-1 flex rounded-md shadow-sm"
      >
        <div
          class="relative flex flex-grow items-stretch focus-within:z-10"
        >
          <input
            autocomplete="off"
            class="block w-full min-w-[200px] rounded-md border-gray-300 py-2 px-2 outline-none hover:cursor-pointer sm:text-sm"
            id="search"
            name="search"
            placeholder="Search by keyword"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="gender"
      >
        Gender
      </label>
      <select
        class="mt-1 block w-full min-w-[200px] rounded-md border-gray-300 bg-white py-2 pl-3 pr-10 text-base shadow-sm outline-none hover:cursor-pointer sm:text-sm"
        id="gender"
        name="gender"
        placeholder=""
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
    <button
      class="font-small items-center rounded-md border border-blue-300 bg-blue-200 px-2 py-2 text-sm text-blue-500 hover:bg-blue-100 hover:text-blue-400"
      type="button"
    >
      Reset Filter
    </button>
  </div>
  <div
    class="flex flex-col"
  >
    <div
      class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
      >
        <div
          class="overflow-hidden border-b border-gray-200 shadow sm:rounded-lg"
        >
          <div
            id="table-container"
          >
            <table
              class="min-w-full divide-y divide-neutral-400"
            >
              <thead
                class="bg-blue-500"
              >
                <tr>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Username
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Name
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Email
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Gender
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Registered Date
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
            </table>
            <div
              class="flex h-80 w-full items-center justify-center"
            >
              <span
                class="text-xl"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 flex justify-end"
    >
      <ul
        class="flex"
      >
        <li>
          <button
            class="inline-flex items-center rounded-l-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="z-10 border-blue-500 bg-blue-100 text-blue-600 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            4
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            5
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center rounded-r-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render UserList renders correctly with simple props: 1`] = `
<DocumentFragment>
  <span
    class="hover:text-blue-400"
  >
    Home
  </span>
  <span
    class="px-2"
  >
    /
  </span>
  <h2
    class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl"
  >
    User List
  </h2>
  <div
    class="flex flex-col space-y-4 md:flex-row md:items-end md:space-y-0 md:space-x-4"
  >
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="search"
      >
        Search
      </label>
      <div
        class="mt-1 flex rounded-md shadow-sm"
      >
        <div
          class="relative flex flex-grow items-stretch focus-within:z-10"
        >
          <input
            autocomplete="off"
            class="block w-full min-w-[200px] rounded-md border-gray-300 py-2 px-2 outline-none hover:cursor-pointer sm:text-sm"
            id="search"
            name="search"
            placeholder="Search by keyword"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="gender"
      >
        Gender
      </label>
      <select
        class="mt-1 block w-full min-w-[200px] rounded-md border-gray-300 bg-white py-2 pl-3 pr-10 text-base shadow-sm outline-none hover:cursor-pointer sm:text-sm"
        id="gender"
        name="gender"
        placeholder=""
      >
        <option
          value=""
        >
          All
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
    <button
      class="font-small items-center rounded-md border border-blue-300 bg-blue-200 px-2 py-2 text-sm text-blue-500 hover:bg-blue-100 hover:text-blue-400"
      type="button"
    >
      Reset Filter
    </button>
  </div>
  <div
    class="flex flex-col"
  >
    <div
      class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
      >
        <div
          class="overflow-hidden border-b border-gray-200 shadow sm:rounded-lg"
        >
          <div
            id="table-container"
          >
            <table
              class="min-w-full divide-y divide-neutral-400"
            >
              <thead
                class="bg-blue-500"
              >
                <tr>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Username
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Name
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Email
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Gender
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Registered Date
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
            </table>
            <div
              class="flex h-80 w-full items-center justify-center"
            >
              <span
                class="text-xl"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 flex justify-end"
    >
      <ul
        class="flex"
      >
        <li>
          <button
            class="inline-flex items-center rounded-l-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="z-10 border-blue-500 bg-blue-100 text-blue-600 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            4
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            type="button"
          >
            5
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center rounded-r-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
